python/demo/cahn-hilliard/demo_cahn-hilliard.py

```diff
@@ -323,7 +266,7 @@ def vector(self):
     T = 50 * dt
 
 u.vector.copy(result=u0.vector)
-u0.vector.ghostUpdate(addv=PETSc.InsertMode.INSERT, mode=PETSc.ScatterMode.FORWARD)
+scatter_forward(u.x)
```

Comment:
Where did this function come from?

Reply:
This is the function from IndexMap used to update Ghost (introduced by @chrisrichardson ). It is nicer than using PETSc directly, and faster.

Reply:
@garth-wells I've removed most of the documentation from the Pybind layer. In general we need to think about how we can customize the underlying linear solver. However, I feel that is out of the scope of this PR.